File: php/src/Behavioral/Template_Method/DvdTitleInfo.php
<?

#//DvdTitleInfo - one of three concrete templates

require_once 'TitleInfo.php';

class DvdTitleInfo extends TitleInfo
{

	private $star;
	private $region;

	function __construct($titleName,$star,$region)
	{
		$this->setTitleName($titleName);
		$this->star = $star;
		$this->region = $region;
	}

	function setStar($s) { $this->star($s); }
	function getStar() {	return $this->star; }
	function setRegion($r) { $this->region = $r; }
	function getRegion(){	return $this->region; }
		
	function getTitleBlurb()
	{
		return "DVD: " . $this->getTitleName() .  ", starring " . $this->getStar();
	}
}

File: go/behavioral/templatemethod/DvdTitleInfo.go
package templatemethod


//DvdTitleInfo.java - one of three concrete templates

type DvdTitleInfo struct { // implements TitleInfoInterface, has TitleInfoImpl
	titleInfo *TitleInfoImpl
   star string 
   encodingRegion string
}
    
func NewDvdTitleInfo(titleName, star, encodingRegion string) *DvdTitleInfo {
	obj := new(DvdTitleInfo)
	obj.titleInfo = new(TitleInfoImpl)
	obj.titleInfo.SetTitleName(titleName)
	obj.star = star
	obj.encodingRegion = encodingRegion
	return obj
}
   
func (this *DvdTitleInfo) GetTitleBlurb() string {
	return "DVD: " + this.titleInfo.GetTitleName() + ", starring " + this.star
}
   
func (this *DvdTitleInfo) GetDvdEncodingRegionInfo() string {
	return ", encoding region: " + this.encodingRegion
}
